python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -500,24 +500,87 @@ class D(A, C): pass
       .isEqualTo("B");
   }
 
+  @Test
+  void annotatedClassFieldsInClassDefinition() {
+    var expression = lastExpression(
+      """
+        class A:
+          a : str = "hi"
+          b : int
+        A
+        """
+    );
+    assertThat(expression.typeV2()).isInstanceOfSatisfying(ClassType.class, classType -> {
+      assertThat(classType.members()).hasSize(2);
+      var aFieldMember = classType.members().stream().filter(m -> "a".equals(m.name())).findFirst().get();
+      var bFieldMember = classType.members().stream().filter(m -> "b".equals(m.name())).findFirst().get();
+      assertThat(aFieldMember.name()).isEqualTo("a");
+      assertThat(aFieldMember.type()).isInstanceOf(ObjectType.class).extracting(PythonType::unwrappedType).isEqualTo(STR_TYPE);
+      assertThat(bFieldMember.name()).isEqualTo("b");
+      assertThat(bFieldMember.type()).isInstanceOf(ObjectType.class).extracting(PythonType::unwrappedType).isEqualTo(INT_TYPE);
+    });
+  }
+
+  @Test
+  void annotatedClassFieldsInClassDefinitionSymbol() {
+    var tree = parseWithoutSymbols(
+      """
+        class A:
+          a : str = "hi"
+          b : int
+        """
+    );
+    var projectLevelSymbolTable = new ProjectLevelSymbolTable();
+    projectLevelSymbolTable.addModule(tree, "", pythonFile("mod.py"));
+
+    var symbol = (ClassSymbol) projectLevelSymbolTable.getSymbol("mod.A");
+    var aFieldSymbol = symbol.resolveMember("a").get();
+    var bFieldSymbol = symbol.resolveMember("b").get();
+    assertThat(aFieldSymbol.name()).isEqualTo("a");
+    assertThat(aFieldSymbol.annotatedTypeName()).isEqualTo("str");
+    assertThat(bFieldSymbol.name()).isEqualTo("b");
+    assertThat(bFieldSymbol.annotatedTypeName()).isEqualTo("int");
+  }
+
+
   @Test
   void staticFieldsInClassDefinition() {
     Expression expr = lastExpression("""
       class A:
         test = "hi"
-      A.test
+      A
       """);
-    assertThat(expr.typeV2())
-      .isInstanceOf(UnknownType.class);
+    assertThat(expr.typeV2()).isInstanceOfSatisfying(ClassType.class, classType -> {
+      assertThat(classType.members()).hasSize(1);
+      var member = classType.members().iterator().next();
+      assertThat(member.name()).isEqualTo("test");
+      assertThat(member.type()).isEqualTo(PythonType.UNKNOWN);
+    });
 
     Expression expr2 = lastExpression("""
       class A:
         test = "hi"
         test = True
```

Comment:
I suggest that you update the test with type hint to ensure that the type of `test` Symbol is `bool` in the end.
```suggestion
      class A:
        test : str = "hi"
        test : bool = True
```

Reply:
test added